### PR TITLE
Updated the BackColorChanged and BackgroundImageChanged event test cases in DateTimePickerTest.cs

### PR DESCRIPTION
**Pull request by John-Qiao**
related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Updated the BackColorChanged and BackgroundImageChanged event test cases in DateTimePickerTest.cs

> 1 DateTimePicker_BackColorChangedEvent_Raised_Success()
> 2 DateTimePicker_BackgroundImageChangedEvent_Raised_Success()

## Customer Impact

- None

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Unit tests

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11106)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DateTimePickerTests.cs [hunk 1]
```diff
@@ -275,27 +275,50 @@ public void DateTimePicker_CalendarMonthBackground_SetInvalid_ThrowsArgumentExce
...
+        EventHandler handler = (sender, e) =>
+        {
+            sender.Should().Be(control);
+            e.Should().Be(EventArgs.Empty);
+            callCount++;
+        };
 
-        control.BackgroundImageChanged += (sender, args) => eventWasRaised = true;
+        control.BackgroundImageChanged += handler;
         using (Bitmap bmp = new Bitmap(10, 10))
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DateTimePickerTests.cs:308]: Please use target type new
```suggestion
        using (Bitmap bmp = new(10, 10))
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DateTimePickerTests.cs [hunk 2]
```diff
@@ -275,27 +275,50 @@ public void DateTimePicker_CalendarMonthBackground_SetInvalid_ThrowsArgumentExce
...
         {
             control.BackgroundImage = bmp;
         }
 
-        eventWasRaised.Should().BeTrue();
+        callCount.Should().Be(1);
+
+        control.BackgroundImageChanged -= handler;
+        using (Bitmap bmp = new Bitmap(20, 20))
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DateTimePickerTests.cs:316]:
```suggestion
        using (Bitmap bmp = new(20, 20))
```

**Review by lonitra — APPROVED** [review]
LGTM, just a small suggestion on using target type new

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Review by lonitra — APPROVED** [review]

**Pull request merged**